packages/in_app_purchase/in_app_purchase_android/lib/src/types/google_play_product_details.dart

```diff
@@ -11,35 +11,135 @@ import '../../billing_client_wrappers.dart';
 class GooglePlayProductDetails extends ProductDetails {
   /// Creates a new Google Play specific product details object with the
   /// provided details.
-  GooglePlayProductDetails({
+  GooglePlayProductDetails._({
     required super.id,
     required super.title,
     required super.description,
     required super.price,
     required super.rawPrice,
     required super.currencyCode,
-    required this.skuDetails,
+    required this.productDetails,
     required super.currencySymbol,
+    this.subscriptionIndex,
   });
 
   /// Generate a [GooglePlayProductDetails] object based on an Android
-  /// [SkuDetailsWrapper] object.
-  factory GooglePlayProductDetails.fromSkuDetails(
-    SkuDetailsWrapper skuDetails,
+  /// [ProductDetailsWrapper] object for an in-app product.
+  factory GooglePlayProductDetails._fromOneTimePurchaseProductDetails(
+    ProductDetailsWrapper productDetails,
   ) {
-    return GooglePlayProductDetails(
-      id: skuDetails.sku,
-      title: skuDetails.title,
-      description: skuDetails.description,
-      price: skuDetails.price,
-      rawPrice: skuDetails.priceAmountMicros / 1000000.0,
-      currencyCode: skuDetails.priceCurrencyCode,
-      currencySymbol: skuDetails.priceCurrencySymbol,
-      skuDetails: skuDetails,
+    assert(productDetails.productType == ProductType.inapp);
+    assert(productDetails.oneTimePurchaseOfferDetails != null);
+
+    final OneTimePurchaseOfferDetailsWrapper oneTimePurchaseOfferDetails =
+        productDetails.oneTimePurchaseOfferDetails!;
+
+    final String formattedPrice = oneTimePurchaseOfferDetails.formattedPrice;
+    final double rawPrice =
+        oneTimePurchaseOfferDetails.priceAmountMicros / 1000000.0;
+    final String currencyCode = oneTimePurchaseOfferDetails.priceCurrencyCode;
+    final String currencySymbol =
+        formattedPrice.isEmpty ? currencyCode : formattedPrice[0];
```

Comment:
Is it guaranteed that the currency symbol is exactly the first character in all cases?

Reply:
I checked and it's not, thank you for noticing. I will employ some regex to catch all* cases.